plugins/uefi-dbx/fu-uefi-dbx-plugin.c

```diff
@@ -38,7 +49,28 @@ fu_uefi_dbx_plugin_coldplug(FuPlugin *plugin, FuProgress *progress, GError **err
 		fu_device_inhibit(FU_DEVICE(device),
 				  "no-dbx",
 				  "System firmware cannot accept DBX updates");
+		inhibited = TRUE;
+	}
+
+#ifdef WITH_UEFI_DBX_SNAPD_NOTIFIER
+	/* TODO progress? */
+	if (self->snapd_notifier != NULL) {
+		g_debug("adding snapd notifier");
+		fu_uefi_dbx_device_set_snapd_notifier(device, self->snapd_notifier);
+	} else if (!inhibited && self->snapd_integration_supported && fu_snap_is_in_snap()) {
+		/* we're running inside a snap, the device is not inhibited and snapd
+		 * supports integration, in which case this is a hard error and we
+		 * should not give an option to dbx */
```

Comment:
if you're running in the snap even on an old snapd without the integration, do you want to inhibit perhaps?  It sounds like it's not a good scenario for updating DBX in that case too I'd think.

Reply:
I think it should only be made a hard-error only if Ubuntu FDE is detected. I left a TODO about it in `fu_common_check_full_disk_encryption()`, which is hopefully the next bit that I may be working on after finishing the dbx work in fwupd and snapd.